pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -53,16 +55,32 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
 
     _readersByIndex = new HashMap<>();
     for (IndexType<?, ?, ?> indexType : IndexService.getInstance().getAllIndexes()) {
-      if (segmentReader.hasIndexFor(columnName, indexType)) {
-        IndexReaderFactory<?> readerProvider = indexType.getReaderFactory();
-        try {
-          IndexReader reader = readerProvider.createIndexReader(segmentReader, fieldIndexConfigs, metadata);
-          if (reader != null) {
-            _readersByIndex.put(indexType, reader);
-          }
-        } catch (IndexReaderConstraintException ex) {
-          LOGGER.warn("Constraint violation when indexing " + columnName + " with " + indexType + " index", ex);
+      Map<String, Map<String, String>> columnProperties = indexLoadingConfig.getColumnProperties();
+      if (!indexType.getId().equals(StandardIndexes.TEXT_ID)) {
```

Comment:
This should be:

```
    for (IndexType<?, ?, ?> indexType : IndexService.getInstance().getAllIndexes()) {
      Map<String, Map<String, String>> columnProperties = indexLoadingConfig.getColumnProperties();
      if (!indexType.getId().equals(StandardIndexes.TEXT_ID)) {
        // process all index types other than Text Index as-it-is
        prepareIndexReader(segmentReader, indexType, fieldIndexConfigs, metadata);
      } else if (IndexLoadingConfig.processExistingSegments(columnName, columnProperties) || segmentReader.hasIndexFor(columnName, StandardIndexes.TEXT_ID)) {
        // In case of Text Index, process segments only if property allows it OR text index exists on disk
        prepareIndexReader(segmentReader, indexType, fieldIndexConfigs, metadata);
      } else {
        LOGGER.info("skipping text index reader for segmentDir: {} for column: {} with skipExistingSegments.",
                segmentReader.toSegmentDirectory().getIndexDir().toString(), columnName);
      }
    }
```